java/ql/lib/semmle/code/java/security/IntentUriPermissionManipulation.qll

```diff
@@ -123,3 +132,12 @@ private predicate bitwiseCheck(Guard g, boolean branch) {
     else g.(EqualityTest).polarity().booleanNot() = branch
   )
 }
+
+/**
+ * Holds if taint can flow from `source` to `sink` in one local step,
+ * including bitwise operations.
+ */
+private predicate bitwiseLocalTaintStep(Expr source, Expr sink) {
```

Comment:
This is almost, but not quite, what you want. Since you've expressed the step relation in terms of `Expr` rather than `DataFlow::Node` then you're ruling out paths that use non-`Expr` intermediate nodes.

Reply:
Hmmm. Does `TaintTracking::localExprTaint` have the same problem then?

Reply:
No, take a look at how it's defined, it restricts the endpoints of the path to be `Expr` - not the intermediate nodes.

Reply:
Done in 3957ebe. This PR was full of traps I set to myself! 😅